tiledb/sm/c_api/tiledb_filestore.cc

```diff
@@ -243,9 +245,16 @@ TILEDB_EXPORT int32_t tiledb_filestore_uri_import(
   std::istream input(&fb);
 
   uint64_t tile_extent = compute_tile_extent_based_on_file_size(file_size);
+  auto&& [st3, buffer_size] = get_buffer_size_from_config(context, tile_extent);
+  if (!st3.ok()) {
+    LOG_STATUS(st3);
+    save_error(ctx, st3);
+    return TILEDB_ERR;
+  }
+
   Query query(context, array);
   query.set_layout(TILEDB_GLOBAL_ORDER);
-  std::vector<std::byte> buffer(tile_extent);
+  std::vector<std::byte> buffer(*buffer_size);
```

Comment:
You need to round the buffer size to a tile extent

Reply:
That is rounded in the function computing the buffer size.